homeassistant/components/media_player/plex.py

```diff
@@ -272,93 +646,309 @@ def _active_media_plexapi_type(self):
+            features = (SUPPORT_PAUSE | SUPPORT_PREVIOUS_TRACK |
+                        SUPPORT_NEXT_TRACK | SUPPORT_STOP |
+                        SUPPORT_VOLUME_SET | SUPPORT_PLAY | SUPPORT_SEEK |
+                        SUPPORT_TURN_OFF | SUPPORT_VOLUME_MUTE)
+
+        # only show controls when we know what device is connecting
+        if not self._make:
+            features = None
+        # no mute support
+        elif self.make.lower() == "shield android tv":
+            _LOGGER.debug(
+                'Shield Android TV client detected, disabling mute '
+                'controls: %s', self.entity_id)
+            features = (SUPPORT_PAUSE | SUPPORT_PREVIOUS_TRACK |
+                        SUPPORT_NEXT_TRACK | SUPPORT_STOP |
+                        SUPPORT_VOLUME_SET | SUPPORT_PLAY | SUPPORT_SEEK |
+                        SUPPORT_TURN_OFF)
+        # Only supports play,pause,stop (and off which really is stop)
+        elif self.make.lower().startswith("tivo"):
+            _LOGGER.debug(
+                'Tivo client detected, only enabling pause, play, '
+                'stop, and off controls: %s', self.entity_id)
+            features = (SUPPORT_PAUSE | SUPPORT_PLAY | SUPPORT_STOP |
+                        SUPPORT_TURN_OFF)
+        # Default to show all controls
+        elif self.device:
+            features = (SUPPORT_PAUSE | SUPPORT_PREVIOUS_TRACK |
+                        SUPPORT_NEXT_TRACK | SUPPORT_STOP |
+                        SUPPORT_VOLUME_SET | SUPPORT_PLAY | SUPPORT_SEEK |
+                        SUPPORT_TURN_OFF | SUPPORT_VOLUME_MUTE)
+        else:
+            features = None
+
+        return features
+
+    def _local_client_control_fix(self):
+        """Detect if local client and adjust url to allow control."""
+        if self.device is None:
+            return
+
+        # if this device's machineIdentifier matches an active client
+        # with a loopback address, the device must be local or casting
+        for client in self.device.server.clients():
+            if ("127.0.0.1" in client.baseurl and
+                    client.machineIdentifier == self.device.machineIdentifier):
+                # point controls to server since that's where the
+                # playback is occuring
+                _LOGGER.debug(
+                    'Local client detected, redirecting controls to '
+                    'Plex server: %s', self.entity_id)
+                server_url = self.device.server.baseurl
+                client_url = self.device.baseurl
+                self.device.baseurl = "{}://{}:{}".format(
+                    urlparse(client_url).scheme,
+                    urlparse(server_url).hostname,
+                    str(urlparse(client_url).port))
 
     def set_volume_level(self, volume):
         """Set volume level, range 0..1."""
-        self.device.setVolume(int(volume * 100),
-                              self._active_media_plexapi_type)
+        if self.device:
```

Comment:
I added the following to all control commands:
`if self.device and 'playback' in self._device_protocol_capabilities:`

I'll commit later after I test